### DISCUSSION TITLE
How to disable pressing F5

### DISCUSSION THREAD

**Discussion by linesoft2** [tauri-apps/tauri#3844] [Q&A] [answered]
On Windows

**Comment by FabianLars** [comment, marked as answer]
Currently you need to overwrite the f5 event via `addEventListener` (including the `event.preventDefault()`).
Furthermore you probably would need to disable the `contextmenu` event too since the menu has a `reload` option.

If you need a context menu you'd have to create one with html/js, therefore i generally recommend making your app "reload-proof" for example by always caching relevant values.

**Reply by linesoft2** [reply]
I think you should use it as the default setting, like Electron.

**Reply by FabianLars** [reply]
While i do agree in general, it's a bit tricky for us to change it since the WebViews don't have good built-in options other than the key events :/

**Reply by dolev146** [reply]
@FabianLars @linesoft2  Hi guys I want to understand the solution

you meant to this snippet right? 
```
document.addEventListener('keydown', function(event) {
  // Prevent F5 or Ctrl+R (Windows/Linux) and Command+R (Mac) from refreshing the page
  if (event.key === 'F5' || (event.ctrlKey && event.key === 'r') || (event.metaKey && event.key === 'r')) {
    event.preventDefault();
  }
});
```

can i also disable the right click with the mouse? 

![image](https://github.com/tauri-apps/tauri/assets/62290677/0f375c9a-7fd3-4cd3-97bc-ede94d4ebf1d)

**Reply by FabianLars** [reply]
This is the "contextmenu" event i mentioned above. 
```js
document.addEventListener('contextmenu', function(event) {
    event.preventDefault();
});
```

**Reply by ddublon** [reply]
@FabianLars  so if i want to disable the refresh in my tauri app then i need to call this function right?

```
const disableRefresh = () => {
  document.addEventListener('keydown', function (event) {
    // Prevent F5 or Ctrl+R (Windows/Linux) and Command+R (Mac) from refreshing the page
    if (
      event.key === 'F5' ||
      (event.ctrlKey && event.key === 'r') ||
      (event.metaKey && event.key === 'r')
    ) {
      event.preventDefault();
    }
  });

  document.addEventListener('contextmenu', function (event) {
    event.preventDefault();
  });
};

```

**Reply by FabianLars** [reply]
I'd say so, yes.

**Reply by dolev146** [reply]
@FabianLars 

fabian is there a possibility to remove only this 2 from the menu instead?

![image](https://github.com/tauri-apps/tauri/assets/62290677/ea70c742-da39-4e72-a063-f7c747a6faaf)

**Reply by FabianLars** [reply]
Possible yes, but it's not exactly easy as it has to happen in rust via [with_webview](https://docs.rs/tauri/latest/tauri/window/struct.Window.html#method.with_webview): https://learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/context-menus?tabs=csharp#removing-menu-items-from-a-default-context-menu
